src/lib.rs

```diff
@@ -167,6 +171,32 @@ impl Resource {
     }
 }
 
+impl std::str::FromStr for Resource {
+    type Err = RlimitsError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "RLIMIT_AS" => Ok(Resource::AS),
+            "RLIMIT_CORE" => Ok(Resource::CORE),
+            "RLIMIT_CPU" => Ok(Resource::CPU),
+            "RLIMIT_DATA" => Ok(Resource::DATA),
+            "RLIMIT_FSIZE" => Ok(Resource::FSIZE),
+            "RLIMIT_LOCKS" => Ok(Resource::LOCKS),
+            "RLIMIT_MEMLOCK" => Ok(Resource::MEMLOCK),
+            "RLIMIT_MSGQUEUE" => Ok(Resource::MSGQUEUE),
+            "RLIMIT_NICE" => Ok(Resource::NICE),
+            "RLIMIT_NOFILE" => Ok(Resource::NOFILE),
+            "RLIMIT_NPROC" => Ok(Resource::NPROC),
+            "RLIMIT_RSS" => Ok(Resource::RSS),
+            "RLIMIT_RTPRIO" => Ok(Resource::RTPRIO),
+            "RLIMIT_RTTIME" => Ok(Resource::RTTIME),
+            "RLIMIT_SIGPENDING" => Ok(Resource::SIGPENDING),
+            "RLIMIT_STACK" => Ok(Resource::STACK),
+            _ => Err(format!("invalid rlimit: {}", s).into()),
+        }
+    }
```

Comment:
We should take `cfg target` into consideration.
By the way, I'm not sure cfg settings in `src/lib.rs` are correct or not.